src/ethereum/ethEngine.ts

```diff
@@ -647,6 +648,23 @@ export class EthereumEngine extends CurrencyEngine<
     await this.startEngine()
   }
 
+  async getFreshAddress(_options: any): Promise<EdgeFreshAddress> {
+    const legacyAddress = this.walletLocalData.publicKey.toLowerCase()
+
+    const { publicKey } = this.walletLocalData
+    const publicAddress = /[A-F]/.test(publicKey)
+      ? publicKey
+      : ((): string => {
+          const addressHex = publicKey.replace('0x', '')
+          return EthereumUtil.toChecksumAddress(addressHex)
+        })()
```

Comment:
This IIFE isn't needed. If you do it the "obvious" way, `? publicKey : EthereumUtil.toChecksumAddress(publicKey.replace('0x', ''))`, the right-hand-side won't be evaluated if the ternary fails.

If this is about formatting, you can just move the `const addressHex = publicKey.replace('0x', '')` line outside the ternary, and not worry about it too much. It's going to be pretty cheap, even if not needed.